lib/rubocop/target_ruby.rb

```diff
@@ -192,14 +192,40 @@ def version_from_gemspec_file(file)
         required_ruby_version(processed_source.ast).first
       end
 
+      def version_from_right_hand_side(right_hand_side)
+        if right_hand_side.array_type?
+          version_from_array(right_hand_side)
+        elsif gem_requirement?(right_hand_side)
+          remove_patch_from_version(right_hand_side.children.last.value)
+        else
+          remove_patch_from_version(right_hand_side.value)
+        end
+      end
+
       def version_from_array(array)
-        versions = array.children.map { |v| version_from_str(v.is_a?(String) ? v : v.str_content) }
-        versions.compact.min
+        array.children.map { |v| remove_patch_from_version(v.value) }
       end
 
-      def version_from_str(str)
-        str.match(/^(?:>=|<=)?\s*(?<version>\d+(?:\.\d+)*)/) do |md|
-          md[:version].to_f
+      def remove_patch_from_version(str)
```

Comment:
I see that you went through the trouble of parsing the requirement nevertheless... How about instead mapping our versions like 2.7 => '2.7.999'? This would have the desired effect, except in the case of exact Ruby version requirements (which are a mistake), simplify the code and avoid redoing the parsing that we shouldn't be doing.